Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.MapViewModel.swift

```diff
@@ -27,19 +27,31 @@ extension OfflineMapAreasView {
         /// The offline map task.
         private let offlineMapTask: OfflineMapTask
         
-        /// The preplanned offline map information.
+        /// The preplanned map information.
         @Published private(set) var preplannedMapModels: Result<[PreplannedMapModel], Error>?
         
+        /// The offline preplanned map information sourced from downloaded mobile map packages.
+        @Published private(set) var offlinePreplannedMapModels: [PreplannedMapModel]?
+        
         init(map: Map) {
             offlineMapTask = OfflineMapTask(onlineMap: map)
             portalItemID = map.item?.id
         }
         
-        /// Gets the preplanned map areas from the offline map task and creates the
-        /// offline map models.
-        func makePreplannedOfflineMapModels() async {
+        /// Requests authorization to show notifications.
+        nonisolated func requestUserNotificationAuthorization() async {
+            _ = try? await UNUserNotificationCenter.current()
+                .requestAuthorization(options: [.alert, .sound])
+        }
+        
+        /// Gets the preplanned map areas from the offline map task and creates the map models.
+        func makePreplannedMapModels() async {
             guard let portalItemID else { return }
             
+            if offlineMapTask.loadStatus == .failed {
```

Comment:
With the recent addition of `NWMonitor`, I feel that it is unlikely the `makePreplannedMapModels()` method would be called when the app is offline. However, I think it may be worth keeping this just as a sanity check. 

Can you think of edge cases that require this?

Reply:
No I couldn't think of an edge case to require this, but I left this in as a sanity check.